File: tool/generate_icon.dart
import 'dart:io';
import 'dart:ui' as ui;
import 'package:flutter/material.dart';

Future<void> main() async {
  WidgetsFlutterBinding.ensureInitialized();
  
  // Criar ícone principal
  await createIcon(
    'assets/icon/app_icon.png',
    size: 1024,
    withPadding: true,
  );
  
  // Criar ícone foreground para adaptive icon
  await createIcon(
    'assets/icon/app_icon_foreground.png',
    size: 1024,
    withPadding: false,
  );
  
  debugPrint('✅ Ícones criados com sucesso!');
  exit(0);
}

Future<void> createIcon(String path, {required double size, required bool withPadding}) async {
  final recorder = ui.PictureRecorder();
  final canvas = Canvas(recorder);
  
  final iconSize = withPadding ? size * 0.6 : size * 0.8;
  final offset = withPadding ? size * 0.2 : size * 0.1;
  
  // Desenhar fundo (apenas para ícone com padding)
  if (withPadding) {
    final bgPaint = Paint()
      ..shader = ui.Gradient.linear(
        const Offset(0, 0),
        Offset(size, size),
        const [Color(0xFF42A5F5), Color(0xFF1976D2)],
      );
    
    final rrect = RRect.fromRectAndRadius(
      Rect.fromLTWH(0, 0, size, size),
      const Radius.circular(0.2),
    );
    canvas.drawRRect(rrect, bgPaint);
  }
  
  // Desenhar ícone de fórum
  // Balão de conversa principal
  final mainBalloonRect = RRect.fromRectAndRadius(
    Rect.fromLTWH(
      offset,
      offset,
      iconSize * 0.7,
      iconSize * 0.6,
    ),
    Radius.circular(iconSize * 0.1),
  );
  
  final iconPaint = Paint()
    ..color = Colors.white
    ..style = PaintingStyle.fill;
  
  canvas.drawRRect(mainBalloonRect, iconPaint);
  
  // Triângulo do balão (pontinho)
  final trianglePath = Path();
  trianglePath.moveTo(offset + iconSize * 0.15, offset + iconSize * 0.6);
  trianglePath.lineTo(offset + iconSize * 0.05, offset + iconSize * 0.75);
  trianglePath.lineTo(offset + iconSize * 0.25, offset + iconSize * 0.6);
  trianglePath.close();
  canvas.drawPath(trianglePath, iconPaint);
  
  // Balão de conversa secundário (menor, atrás)
  final secondBalloonRect = RRect.fromRectAndRadius(
    Rect.fromLTWH(
      offset + iconSize * 0.4,
      offset + iconSize * 0.25,
      iconSize * 0.55,
      iconSize * 0.5,
    ),
    Radius.circular(iconSize * 0.08),
  );
  
  final secondPaint = Paint()
    ..color = const Color.fromRGBO(255, 255, 255, 0.9)
    ..style = PaintingStyle.fill;
  
  canvas.drawRRect(secondBalloonRect, secondPaint);
  
  // Linhas de texto no balão principal
  final linePaint = Paint()
    ..color = const Color(0xFF1976D2)
    ..strokeWidth = iconSize * 0.02
    ..strokeCap = StrokeCap.round;
  
  // Linha 1
  canvas.drawLine(
    Offset(offset + iconSize * 0.15, offset + iconSize * 0.25),
    Offset(offset + iconSize * 0.55, offset + iconSize * 0.25),
    linePaint,
  );
  
  // Linha 2
  canvas.drawLine(
    Offset(offset + iconSize * 0.15, offset + iconSize * 0.38),
    Offset(offset + iconSize * 0.50, offset + iconSize * 0.38),
    linePaint,
  );
  
  final picture = recorder.endRecording();
  final image = await picture.toImage(size.toInt(), size.toInt());
  final byteData = await image.toByteData(format: ui.ImageByteFormat.png);
  final buffer = byteData!.buffer.asUint8List();
  
  final file = File(path);
  await file.parent.create(recursive: true);
  await file.writeAsBytes(buffer);
  
  debugPrint('✅ Criado: $path');
}

File: tool/generate_icon.py
"""
Script para gerar ícones do aplicativo Forum - Mobile
Cria ícones com design de fórum/chat
"""

from PIL import Image, ImageDraw, ImageFont
import os

def create_icon(size, output_path, with_background=True):
    """Cria um ícone com design de fórum"""
    # Criar imagem com fundo azul ou transparente
    if with_background:
        img = Image.new('RGB', (size, size), '#2196F3')
    else:
        img = Image.new('RGBA', (size, size), (0, 0, 0, 0))
    
    draw = ImageDraw.Draw(img)
    
    # Desenhar um balão de conversa estilizado
    margin = size // 8
    bubble_rect = [margin, margin, size - margin, size - margin * 1.3]
    
    # Fundo do balão (branco ou com transparência)
    if with_background:
        draw.rounded_rectangle(bubble_rect, radius=size//10, fill='white')
    else:
        draw.rounded_rectangle(bubble_rect, radius=size//10, fill='#2196F3')
    
    # Desenhar três linhas dentro do balão (representando texto)
    line_margin = size // 4
    line_width = size // 15
    line_spacing = size // 8
    line_y_start = size // 3
    
    for i in range(3):
        y = line_y_start + (i * line_spacing)
        line_length = size - (line_margin * 2) - (i * size // 10)
        if with_background:
            draw.rounded_rectangle(
                [line_margin, y, line_margin + line_length, y + line_width],
                radius=line_width//2,
                fill='#1976D2'
            )
        else:
            draw.rounded_rectangle(
                [line_margin, y, line_margin + line_length, y + line_width],
                radius=line_width//2,
                fill='white'
            )
    
    # Desenhar ponta do balão
    tail_points = [
        (size // 2 - margin, bubble_rect[3]),
        (size // 2 - margin * 2, size - margin // 2),
        (size // 2, bubble_rect[3])
    ]
    if with_background:
        draw.polygon(tail_points, fill='white')
    else:
        draw.polygon(tail_points, fill='#2196F3')
    
    # Salvar
    os.makedirs(os.path.dirname(output_path), exist_ok=True)
    img.save(output_path)
    print(f"Ícone criado: {output_path}")

def main():
    # Criar diretório de assets
    base_path = os.path.join(os.path.dirname(__file__), '..', 'assets', 'icon')
    
    # Criar ícone principal (1024x1024)
    create_icon(1024, os.path.join(base_path, 'app_icon.png'), with_background=True)
    
    # Criar ícone foreground para adaptive icon (1024x1024)
    create_icon(1024, os.path.join(base_path, 'app_icon_foreground.png'), with_background=False)
    
    print("\n✅ Ícones criados com sucesso!")
    print("Execute agora: flutter pub get")
    print("Depois execute: flutter pub run flutter_launcher_icons")

if __name__ == '__main__':
    main()
